server/api/controllers/article.js

```diff
@@ -580,3 +579,120 @@ export const getArticle = async (req, res) => {
     });
   }
 };
+
+/**
+ * @export
+ * @function updateArticle
+ * @param {Object} req - request received
+ * @param {Object} res - response object
+ * @returns {Object} JSON object (JSend format)
+ */
+export const updateArticle = async (req, res) => {
+  try {
+    const {
+      body,
+      params: {
+        articleId
+      },
+      user: {
+        id: userId
+      }
+    } = req;
+
+    const foundArticle = await Article.findOne({
+      where: {
+        id: {
+          [Op.eq]: articleId
+        },
+      }
+    });
+
+    if (!foundArticle) {
+      return res.status(404).send({
+        status: 'fail',
+        message: 'Article not found'
+      });
+    }
+    const articleAuthor = await Article.findOne({
+      where: {
+        id: {
+          [Op.eq]: articleId
+        },
+        authorId: {
+          [Op.eq]: userId
+        }
+      }
+    });
+    if (!articleAuthor) {
+      return res.status(401).send({
+        status: 'fail',
+        message: "Sorry you can't edit this article"
+      });
+    }
+    let newArticle = await foundArticle.update(
+      body, { returning: true, plain: true }
+    );
+
+    newArticle = newArticle.toJSON();
+
+    return res.status(201).send({
+      status: 'success',
+      message: 'Yaay! You just created an article',
+      newArticle
+    });
+  } catch (e) {
+    return res.status(502).send({
+      status: 'Error',
+      message: 'OOPS! an error occurred while trying to update this article. log in and try again!'
+    });
+  }
+};
+
+
+/**
+ * @export
+ * @function deleteArticle
+ * @param {Object} req - request received
+ * @param {Object} res - response object
+ * @returns {Object} JSON object (JSend format)
+ */
+export const deleteArticle = async (req, res) => {
+  try {
+    const { body, params: { articleId }, user: { id: userId, role } } = req;
+
+    const foundArticle = await Article.findOne({ where: { id: { [Op.eq]: articleId } } });
```

Comment:
@oyedejipeace Please remove lines `663` to 670`. They are redundant. The first line of action in this controller should be to check if the role of the authenticated user is `admin`. We only intend to go ahead with the request if this check is successful, not the other way round. This check should be something like this:

``` 
if (role !== 'admin')) {
      return res.status(401).send({
        status: 'fail',
        message: "Unauthorised operation"
      });
    }

```

Next  lines should be:

``` 

const article = await Article.findOne({
      where: {
        id: { [Op.eq]: articleId },
        authorId: { [Op.eq]: userId }
      }
    });

if (!article) {
      return res.status(404).send({
        status: 'fail',
        message: "Not found"
      });
}
await article.destroy(   body, { returning: true, plain: true }); 
.....
.....

```